pkg/templates/machinecontroller/machines.go

```diff
@@ -96,13 +96,9 @@ func createMachineDeployment(cluster *kubeoneapi.KubeOneCluster, workerset kubeo
 	}
 
 	return &clusterv1alpha1.MachineDeployment{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: "cluster.k8s.io/v1alpha1",
-			Kind:       "MachineDeployment",
-		},
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: metav1.NamespaceSystem,
-			Name:      fmt.Sprintf("%s-deployment", workerset.Name),
+			Name:      workerset.Name,
```

Comment:
What is going to happen when we upgrade the cluster? Will it recreate the worker nodes with the new name or keep using the old name?

Reply:
In existing clusters MachineDeployment wouldn't be changed.

Reply:
Sounds good.